common/src/main/java/eu/midnightdust/lib/config/MidnightConfig.java

```diff
@@ -305,14 +321,16 @@ public boolean keyPressed(int keyCode, int scanCode, int modifiers) {
         }
         @Override
         public void close() {
-            loadValuesFromJson(modid); cleanup();
+//            loadValuesFromJson(modid);  // TODO: It is meaningless to call `loadValuesFromJson()`
+            // cleanup();    // TODO: I think it is incorrect to call `cleanup()` because it is unreasonable to reset the `EntryInfo` objects in `entries`.
             Objects.requireNonNull(client).setScreen(parent);
         }
-        private void cleanup() {
-            entries.values().forEach(info -> {
-                info.error = null; info.value = null; info.tempValue = null; info.actionButton = null; info.listIndex = 0; info.tab = null; info.inLimits = true;
-            });
-        }
+//        TODO: What the reason to define this method? I don't think it should be defined and called.
```

Comment:
This is intended to free up some RAM and also reset the error messages, so the user does not see an error message for a cancelled change after reopening the config screen.